packages/yarnpkg-builder/sources/tools/makeConfig.ts

```diff
@@ -34,10 +60,12 @@ export const makeConfig = (config: webpack.Configuration) => merge({
         options: {
           compilerOptions: {
             declaration: false,
-            module: `ESNext`,
-            moduleResolution: `node`,
+            module: `ESNext` as any,
+            moduleResolution: `node` as any,
```

Comment:
Why `as any`? If it's because the types expect a literal value, you can use `as const`

Reply:
It's because `ts-loader` incorrectly expects `module` and `moduleResolution` to be `ts.ModuleKind` and `ts.ModuleResolutionKind`. The problem is that those are numeric enums that aren't accepted by `ts-loader`:
```ts
export enum ModuleKind {
        None = 0,
        CommonJS = 1,
        AMD = 2,
        UMD = 3,
        System = 4,
        ES2015 = 5,
        ES2020 = 6,
        ESNext = 99
    }
```
If I do `module: 'ESNext'`, there's a compile error.
If I do `module: ts.ModuleKind.ESNext`, there's a runtime error.
That's why the only solution is either `as any` or a `@ts-ignore`.